src/app.py

```diff
@@ -31,25 +33,41 @@ def update_clients(self):
                 client_secret = config.client_secret,
                 redirect_uri = config.redirect_uri
             )
-            token = oauth.get_cached_token()['access_token']
             try:
+                token = oauth.get_cached_token()['access_token']
                 playlist.update_playlist(spotipy.Spotify(auth=token))
+
+                # reset the users error count if an update was successful
+                database.update_user(id, "error_count", 0)
             except Exception as e:
                 timestamp = dt.now(tz=tz.utc).strftime('%Y-%m-%d %H:%M:%S')
-                message = "Unable to update playlist for: " + id + "\n"
-                message += str(e)
-                print(timestamp + ": " + message)
-                with open(constant.SRC_PATH + '/../error.log', 'a') as f:
-                    f.write(message)
-                    f.write(traceback.format_exc())
+                # message = "Unable to update playlist for: " + id + "\n"
+                message = str(e)
+                message = timestamp + ": " + message
+                # print(message)
+                database.increment_field(id, "error_count")
+                database.update_user(id, "last_error", message)
+                # with open(constant.SRC_PATH + '/../error.log', 'a') as f:
+                #     f.write(message)
+                #     f.write(traceback.format_exc())
+
+                # if a user passes a certain number of errors, delete their 
+                # token, they probably revoked our access
+                if database.get_field(id, "error_count") > constant.ERROR_THRESHOLD:
+                    try:
+                        os.remove(cache_path)
+                    except OSError:
+                        pass
+                    database.remove_user(id)
```

Comment:
what's the point of keeping track of error messages if you delete the column anyway

Reply:
this is true, it stays in the database now but the file gets deleted